pkg/actuators/machine/machine_scope_test.go

```diff
@@ -0,0 +1,293 @@
+	}
+
+	var err error
+	cfg, err := testEnv.Start()
+	g.Expect(err).ToNot(HaveOccurred())
+	g.Expect(cfg).ToNot(BeNil())
+	defer func() {
+		g.Expect(testEnv.Stop()).To(Succeed())
+	}()
+
+	mgr, err := manager.New(cfg, manager.Options{
+		Scheme:             scheme.Scheme,
+		MetricsBindAddress: "0",
+	})
+	g.Expect(err).ToNot(HaveOccurred())
+
+	doneMgr := make(chan struct{})
+	go func() {
+		g.Expect(mgr.Start(doneMgr)).To(Succeed())
+	}()
+	defer close(doneMgr)
+
+	// END: setup test environment
+
+	k8sClient := mgr.GetClient()
+
+	awsCredentialsSecret := stubAwsCredentialsSecret()
+	g.Expect(k8sClient.Create(context.TODO(), awsCredentialsSecret)).To(Succeed())
+
+	userDataSecret := stubUserDataSecret()
+	g.Expect(k8sClient.Create(context.TODO(), userDataSecret)).To(Succeed())
+
+	failedPhase := "Failed"
+
+	providerStatus := &awsproviderv1.AWSMachineProviderStatus{}
+
+	testCases := []struct {
+		name   string
+		mutate func(*machinev1.Machine)
+		expect func(*machinev1.Machine) error
+	}{
+		{
+			name: "Test changing labels",
+			mutate: func(m *machinev1.Machine) {
+				m.ObjectMeta.Labels["testlabel"] = "test"
+			},
+			expect: func(m *machinev1.Machine) error {
+				if m.ObjectMeta.Labels["testlabel"] != "test" {
+					return fmt.Errorf("label \"testlabel\" %q not equal expected \"test\"", m.ObjectMeta.Labels["test"])
+				}
+				return nil
+			},
+		},
+		{
+			name: "Test setting phase",
+			mutate: func(m *machinev1.Machine) {
+				m.Status.Phase = &failedPhase
+			},
+			expect: func(m *machinev1.Machine) error {
+				if m.Status.Phase != nil && *m.Status.Phase == failedPhase {
+					return nil
+				}
+				return fmt.Errorf("phase is nil or not equal expected \"Failed\"")
+			},
+		},
+		{
+			name: "Test setting provider status",
+			mutate: func(m *machinev1.Machine) {
+				instanceID := "123"
+				instanceState := "running"
+				providerStatus.InstanceID = &instanceID
+				providerStatus.InstanceState = &instanceState
+			},
+			expect: func(m *machinev1.Machine) error {
+				providerStatus, err := awsproviderv1.ProviderStatusFromRawExtension(m.Status.ProviderStatus)
+				if err != nil {
+					return fmt.Errorf("unable to get provider status: %v", err)
+				}
+
+				if providerStatus.InstanceID == nil || *providerStatus.InstanceID != "123" {
+					return fmt.Errorf("instanceID is nil or not equal expected \"123\"")
+				}
+
+				if providerStatus.InstanceState == nil || *providerStatus.InstanceState != "running" {
+					return fmt.Errorf("instanceState is nil or not equal expected \"running\"")
+				}
+
+				return nil
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			timeout := 10 * time.Second
+			gs := NewWithT(t)
+
+			machine, err := stubMachine()
+			gs.Expect(err).ToNot(HaveOccurred())
+			gs.Expect(machine).ToNot(BeNil())
+
+			ctx := context.TODO()
+
+			// Create the machine
+			gs.Expect(k8sClient.Create(ctx, machine)).To(Succeed())
+
```

Comment:
I would remove this empty line to join the Create and Delete together to show they're related